### ISSUE TITLE
Cannot get final name for export 'encodeXML' of ./node_modules/entities/lib/esm/index.js

### ISSUE THREAD

**Issue by Munsterberg** [resendlabs/react-email#1015]
### Describe the Bug

Since upgrading to next 14 we can't get our next app to build, seeing the following error:

```
./node_modules/@react-email/components/node_modules/html-to-text/lib/html-to-text.mjs + 24 modules
Cannot get final name for export 'encodeXML' of ./node_modules/entities/lib/esm/index.js

```

### Which package is affected (leave empty if unsure)

@react-email/components

### Link to the code that reproduces this issue

private

### To Reproduce

Create a next 14 project and use the following packages

```
        "@react-email/components": "0.0.11",
        "@react-email/html": "^0.0.6",
        "@react-email/tailwind": "^0.0.12",
        "react-email": "1.9.5",
```

### Expected Behavior

The build to work

### What's your node version? (if relevant)

18.18.0

**Comment by gabrielmfern** [comment]
Taking a look into fixing this right now. Thanks for the issue!

**Comment by gabrielmfern** [comment]
Quick question, are you using the resend sdk? I am wondering if this is a problem with the resend sdk instead of with react-email

**Comment by gabrielmfern** [comment]
> > Quick question, are you using the resend sdk? I am wondering if this is a problem with the resend sdk instead of with react-email
> 
> We are, 2.0.0. Here is the config below with the `from` changed:
> 
> ```
> import { env } from '@/env.mjs';
> import * as React from 'react';
> import { Resend } from 'resend';
> export const resend = new Resend(env.RESEND_API_KEY);
> 
> export const sendEmail = async ({
>     to,
>     subject,
>     template,
> }: {
>     to: string[];
>     subject: string;
>     template: React.ReactElement;
> }) => {
>     await resend.emails.send({
>         from: '---',
>         to,
>         subject,
>         react: template,
>     });
> };
> ```

Can you try something out for me so that I can confirm my theory? Try, instead of using the react option, rendering the email using the `render` function and passing it on the `html` option and see if the error persists.

**Comment by gabrielmfern** [comment]
> @gabrielmfern that fixed the build, no error this time around!

Great, anyone that has this problem can use this as a workaround for now, and I'll get a fix on this ASAP. Thank you for your patience!

**Comment by gabrielmfern** [comment]
Closing this issue as it is a problem with the resend sdk, if anyone has this problem, read the conversation above and use the workaround for now, once the PR is merged and a new release is out you can just update to the new version.

**Comment by FajarAlnito** [comment]
Hello, I have this problem, what is the `render` function mentioned above? Thank You

**Comment by michaelangeloio** [comment]
@gabrielmfern I'm getting this issue _without_ resend. I've upgraded all my `react-email` packages to latest (including canary too):

```
    "@react-email/button": "^0.0.11",
    "@react-email/components": "0.0.12-canary.0",
    "@react-email/html": "^0.0.6",
    "@react-email/render": "0.0.9-canary.2",
    "react-email": "^1.9.5",
```

**Comment by steveninety** [comment]
I'm _still_ getting this error despite...
...using the render (or renderAsync) function
...downgrading my next and react-email dependencies to match OP
...isolating component rendering from sending the email using Resend (in other words, the render function alone causes this issue)

**Comment by gabrielmfern** [comment]
@steveninety Did you try using both the `html` option for the Resend SDK and upgrading it?

**Comment by Loosetooth** [comment]
I also encountered the above error, using nextjs 14 and react-email. (Without using resend.)

I tried to work around it based on the [this comment](https://github.com/vercel/next.js/issues/59344#issuecomment-1846128759).
Adding the following lines to my `next.config.js` file:
```diff
experimental: {
+    serverComponentsExternalPackages: [
+      '@react-email'
+    ]
  },
```

Makes the build succeed locally. The build still fails on vercel however with some other errors.

I'm still very puzzled on why this error happens, but it seems like it has to do with using `react-email` in server actions/server components, correct?
Or at least, that's the impression I get when reading the [nextjs docs on serverComponentsExternalPackages](https://nextjs.org/docs/app/api-reference/next-config-js/serverComponentsExternalPackages).

Any ideas?
